src/assertions/json.rs

```diff
@@ -0,0 +1,227 @@
+    #[track_caller]
+    #[allow(clippy::wrong_self_convention)]
+    fn is_null(self);
+
+    /// Asserts that the [Value] is a [Value::Bool].
+    ///
+    /// Allows the usage of chained assertions on a bool-value (see [`AssertionConnector`]).
+    ///
+    /// # Examples
+    /// ```
+    /// # use serde_json::json;
+    /// # use smoothy::prelude::*;
+    /// #
+    /// let result = json!(true);
+    ///
+    /// assert_that(result).is_boolean().and().is_true();
+    /// ```
+    ///
+    /// # Panics
+    /// When the [Value] is no [Value::Bool]
+    #[track_caller]
+    #[allow(clippy::wrong_self_convention)]
+    fn is_boolean(self) -> AssertionConnector<bool>;
+
+    /// Asserts that the [Value] is a [Value::Number].
+    ///
+    /// Allows the usage of chained assertions on a number-value (see [`AssertionConnector`]).
+    ///
+    /// # Examples
+    /// ```
+    /// # use serde_json::json;
+    /// # use smoothy::prelude::*;
+    /// #
+    /// let result = json!(42);
+    ///
+    /// assert_that(result).is_number().and().equals(42);
+    /// ```
+    ///
+    /// # Panics
+    /// When the [Value] is no [Value::Number]
+    #[track_caller]
+    #[allow(clippy::wrong_self_convention)]
+    fn is_number(self) -> AssertionConnector<Number>;
+
+    /// Asserts that the [Value] is a [Value::String].
+    ///
+    /// Allows the usage of chained assertions on a string-value (see [`AssertionConnector`]).
+    ///
+    /// # Examples
+    /// ```
+    /// # use serde_json::json;
+    /// # use smoothy::prelude::*;
+    /// #
+    /// let result = json!("test");
+    ///
+    /// assert_that(result).is_string().and().equals("test");
+    /// ```
+    ///
+    /// # Panics
+    /// When the [Value] is no [Value::String]
+    #[track_caller]
+    #[allow(clippy::wrong_self_convention)]
+    fn is_string(self) -> AssertionConnector<String>;
+
+    /// Asserts that the [Value] is a [Value::Array].
+    ///
+    /// Allows the usage of chained assertions on an array-value (see [`AssertionConnector`]).
+    ///
+    /// # Examples
+    /// ```
+    /// # use serde_json::json;
+    /// # use smoothy::prelude::*;
+    /// #
+    /// let result = json!([ null, 42, "test" ]);
+    ///
+    /// assert_that(result).is_array().and().contains(42);
+    /// ```
+    ///
+    /// # Panics
+    /// When the [Value] is no [Value::Array]
+    #[track_caller]
+    #[allow(clippy::wrong_self_convention)]
+    fn is_array(self) -> AssertionConnector<Vec<Value>>;
+
+    /// Asserts that the [Value] is a [Value::Object].
+    ///
+    /// Allows the usage of chained assertions on an object-value (see [`AssertionConnector`]).
+    ///
+    /// # Examples
+    /// ```
+    /// # use serde_json::json;
+    /// # use smoothy::prelude::*;
+    /// #
+    /// let result = json!({
+    ///     "test": 42,
+    ///     "other": null,
+    /// });
+    ///
+    /// assert_that(result).is_object().and().get("test").equals(json!(42));
```

Comment:
i would change the assertion to
`assert_that(result).is_object().and().get("test").is_number().and().equals(42)`
because it shows how one can use both the assertions combined

Reply:
Changed